### PR TITLE
Adding dbjson and fetch

### PR DESCRIPTION
**Pull request by Matt-Eva**
Updated the reading to use a db.json, json-server, useEffect, and fetch to more realistically mimic real applications.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,8 +46,9 @@ into it!
...
 ### Setting up our Main Route
 
-To get started, clone down this repo and run `npm install`. Then run `npm start`
-to open the application in your browser.
+To get started, clone down this repo and run `npm install`. Then run `npm run
+server` to start your `json-server` and `npm start` to open the application in
+your browser.
 
 If you open up `src/index.js`, you will see that we are currently rendering our
 `Home` component, which will serve as the homepage of our application. `Home` is
```
Comment (lizbur10) [inline README.md:54]: Need to update to reflect use of db.json on line 56

File: README.md [hunk 2]
```diff
@@ -67,9 +68,18 @@ root.render(<Home />);
 ```jsx
 // Home.js
 import users from "../data";
```
Comment (lizbur10) [inline README.md]:
```suggestion

```

File: README.md [hunk 3]
```diff
@@ -393,7 +413,6 @@ Here's a basic setup for that component:
```
Comment (lizbur10) [inline README.md:410]: Remove sentence "We'll be importing..." on lines 409-10

File: README.md [hunk 4]
```diff
@@ -578,33 +599,56 @@ Note that the key is the parameter we defined in our route, and the value is
 what appears in the URL.
 
 We can now use the data contained in our params object to access the specific
-piece of data we want to display!
+piece of data we want to display! We can interpolate the `id` or our specific
```
Comment (lizbur10) [inline README.md]:
```suggestion
piece of data we want to display! We can interpolate the `id` of the
```

File: README.md [hunk 5]
```diff
@@ -578,33 +599,56 @@ Note that the key is the parameter we defined in our route, and the value is
...
-`fetch` request to grab the specific piece of data you want from your database.
+```jsx
+if(!user.name){
+  return <h1>Loading...</h1>;
+};
+```
 
-Now that we have a way to access the user we want, let's update our
-`UserProfile` component to display information about that user!
+Here's what our full UserProfile component looks like now:
```
Comment (lizbur10) [inline README.md:632]: Suggest adding just before this line:

Finally, let's update the jsx to display the user's name:

```jsx
<h1>{user.name}</h1>
```

**Review by lizbur10 — CHANGES_REQUESTED** [review]
Looks good! Found a few things

**Comment by Matt-Eva** [comment]
Nice catch! I think I've resolved everything.

**Pull request merged**